torchci/lib/bot/pytorchBotHandler.ts

```diff
@@ -169,6 +171,20 @@ The explanation needs to be clear on why this is needed. Here are some good exam
         rebase = false;
       }
       await this.logger.log("merge", extra_data);
+      if (!forceRequested) {
+        let labels: string[] = this.ctx.payload?.issue?.labels.map(
+          (e: any) => e["name"]
+        );
+        if (labels === undefined) {
+          labels = this.ctx.payload?.pull_request?.labels.map(
+          (e: any) => e["name"]
+          );
+        }
+
+        if (labels !== undefined && ! labels.find( x => x === CIFLOW_TRUNK_LABEL)) {
+          await addLabels(this.ctx, [CIFLOW_TRUNK_LABEL])
```

Comment:
Wondering if we might need to inject a delay of some sort here to prevent the trunk workflow initiation from racing with the mergebot's "all existing workflows are green" check

Reply:
Potentially but this fires prior to the merge payload being sent to repository dispatch so hopefully it shouldn't?